stage4.py

```diff
@@ -0,0 +1,73 @@
+from card import Card
+import random
+
+deck = list()
+is_card_identical = False
+while len(deck) < 40:
+    pulled_card = Card()
+    for card in deck:
+        if card == pulled_card:
+            is_card_identical = True
+            break
+    if is_card_identical:
+        is_card_identical = False
+        continue
+    else:
+        deck.append(pulled_card)
+
+random.shuffle(deck)
+
+# - - - - - - - - - - - - - - - - - - - - - - - - -
+
+num_of_players = 0
+while num_of_players not in range(2, 7):
+    num_of_players_as_str = input("Enter number of players (between 2 to 6): ")
+    num_of_players = int(num_of_players_as_str)
+
+# - - - - - - - - - - - - - - - - - - - - - - - - -
+
+cards = dict()
+for player in range(1, num_of_players + 1):
+    cards[player] = list()
+    for j in range(3):
+        cards[player].append(deck.pop())
+
+card_on_table = deck.pop()
+winner_index = 0
+player_turn = 1
+player_has_card = False
+
+while deck:
+    current_player_num_of_cards = len(cards[player_turn])
+    current_player_cards = cards[player_turn]
+    for i in range(current_player_num_of_cards):
```

Comment:
Welcome the most amazing thing in the high-level language (like Java, C# python..)
We almost never work with array index, in case we would like to iterate over the array we just use it with a for

```
for card in current_player_cards:
  ... do something with card...
```